.github/workflows/zcash-params.yml

```diff
@@ -2,7 +2,7 @@ name: zcash-params
 
 on:
   workflow_dispatch:
-  pull_request:
+  pull_request_target:
     path:
       - 'zebra-consensus/src/primitives/groth16/params.rs'
       - 'zebra-consensus/src/chain.rs'
```

Comment:
Why is this generated before merge to #main? Shouldn't we only run this when changes are merged/pushed to #main that change these files, not from a pre-merge PR branch? The workflow_dispatch is fine

Reply:
Considering that we still don't have #3362, I do agree. For this to happen, it should only impact the specific test instance from that PR, and as of now, it would also affect the main branch. This was a complete oversight from my side.